metrics/meter.go

```diff
@@ -182,11 +184,9 @@ func newStandardMeter() *StandardMeter {
 
 // Stop stops the meter, Mark() will be a no-op if you use it after being stopped.
 func (m *StandardMeter) Stop() {
-	m.lock.Lock()
-	stopped := m.stopped
-	m.stopped = true
-	m.lock.Unlock()
-	if !stopped {
+	stopped := atomic.LoadUint32(&m.stopped)
+	atomic.StoreUint32(&m.stopped, 1)
```

Comment:
I think these two ops are semantically equivalent to `stopped := atomic.SwapUint32(&m.stopped, 1)`

Reply:
Yep sounds good!